converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Observation/ObservationParser.cs

```diff
@@ -0,0 +1,354 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Dicom;
+using Dicom.StructuredReport;
+using Hl7.Fhir.Model;
+using Microsoft.Health.DicomCast.Core.Features.Fhir;
+
+namespace Microsoft.Health.DicomCast.Core.Features.Worker.FhirTransaction
+{
+    public static class ObservationParser
+    {
+        /// <summary>
+        /// Parses the DicomStructuredReports in a DicomDataset for the contained Observations.
+        /// </summary>
+        /// <remarks>
+        /// For observation profiles see: https://confluence.hl7.org/display/IMIN/Radiation+Dose+Summary+for+Diagnostic+Procedures+on+FHIR#RadiationDoseSummaryforDiagnosticProceduresonFHIR-DoseSummaryObservationProfile
+        /// </remarks>
+        /// <param name="dataset">The dataset to parse</param>
+        /// <param name="patientRef">Reference to an existing patient to link to</param>
+        /// <param name="imagingStudyRef">Reference to an existing ImagingStudy to link to</param>
+        /// <returns>An object containing the Observations found in the dataset</returns>
+        public static ParsedObservation CreateObservations(
+            DicomDataset dataset,
+            ResourceReference patientRef,
+            ResourceReference imagingStudyRef)
+        {
+            var parsedObservations = new ParsedObservation();
+
+            // run the loop
+            ParseObservations(dataset, patientRef, imagingStudyRef, parsedObservations);
+
+            // Set each observation status to Preliminary
+            foreach (Observation doseSummary in parsedObservations.DoseSummaries)
+            {
+                doseSummary.Status = ObservationStatus.Preliminary;
```

Comment:
Any reason to not just simplify this and change it such that every parsed observation is just automoatcially set with a `Status` of `ObservationStatus.Preliminary`?